snuba/datasets/errors_replacer.py

```diff
@@ -526,8 +541,42 @@ def process_message(self, message: ReplacementMessage) -> Optional[Replacement]:
 
         return processed
 
+    def _message_already_processed(self, message: ReplacementMessage) -> bool:
+        """
+        Figure out whether or not the message was already processed.
+
+        Check whether there is an entry in Redis for this specific
+        topic, consumer group, and partition. If there is, we can conclude whether
+        or not to process the incoming message by comparing the incoming message
+        to the offset in Redis.
+        """
+        key = self._build_topic_group_index_key(message.metadata)
+        processed_offset = redis_client.get(key)
+        if processed_offset is not None:
+            offset = int(processed_offset)
```

Comment:
If this fails you can clean up the redis key (so the replacement does not get stuck). If somebody corrupted the data in Redis and the content is not an int anymore, there is no point in keeping the state around as there is nothing we will be able to do to fix it.  Just clean it up, log an error and go on.